added super-resolution example

VictorIyke opened the pull request:
Added super-resolution example using ORT that could be run on both Android and Web

natke commented on js/app-super-resolution/.gitignore:
```diff
@@ -0,0 +1,75 @@
+node_modules/
+.expo/
```
We should add .expo_shared here too (I think) and then remove the files in .expo_shared from the repo

VictorIyke replied:
👍

natke commented on js/app-super-resolution/README.md:
```diff
@@ -0,0 +1,59 @@
+# RUNNING SUPER-RESOLUTION ON IMAGES ACROSS VARIOUS PLATFORMS USING ONNX RUNTIME MODELS
```
Let's make it clear here that the app runs on web and android and not iOS yet.

And if you know, the steps needed to get it working on iOS

VictorIyke replied:
👍

natke commented on js/app-super-resolution/README.md:
```diff
@@ -0,0 +1,59 @@
+    2. Copy contents of this Repo into created directory
+
+    **NOTE:**
+    - `<SOURCE_ROOT>` refers to the root folder of the source code directory.
```
Is this the root of the onnxruntime-inference-examples, or the app directory?

Do they need to copy, or can the clone the examples repo and change directories into the app folder?

VictorIyke replied:
It refers to the app directory. They can clone the examples repo and change directories into the app folder. I will specify that in the README

natke commented on js/app-super-resolution/README.md:
````diff
@@ -0,0 +1,59 @@
+         yarn
+         ```
+      
+      2. Using NPM:
````
Do we need to do this if we are using expo?

VictorIyke replied:
You can do either. I will specify that

natke reviewed (commented):
Looking good. A couple of comments:
- which of the files in the android folder are actually source files? We should remove any files that get built?
- would there be any way of merging the screens for the mobile and web app more, so that more code is cross platform?

YUNQIUGUO commented on js/app-super-resolution/README.md:
````diff
@@ -0,0 +1,59 @@
+3. Run the following commands to launch:
+
+    In `<SOURCE_ROOT>`, run the following command to launch for Android
+    ```sh
````
nit: the commands for android/web is meant to be different? 

Is there a way to make it more consistent?

VictorIyke replied:
From my knowledge, it should be different as Expo takes some extra steps in order to build the app for Android (also the same for IOS)

YUNQIUGUO commented on js/app-super-resolution/README.md:
```diff
@@ -0,0 +1,59 @@
+# RUNNING SUPER-RESOLUTION ON IMAGES ACROSS VARIOUS PLATFORMS USING ONNX RUNTIME MODELS
```
general suggestion for read.me: 

maybe can make it more complete follow the same pattern/sections from maui example:

https://github.com/microsoft/onnxruntime-inference-examples/blob/main/mobile/examples/Maui/MauiVisionSample/readme.md

maybe for sections like"pre/post processing" methods, can expand to explain in more details in github tutorial page, and briefly talk about it here would be good.

VictorIyke replied:
adding that now

YUNQIUGUO commented on js/app-super-resolution/screens/mobile.tsx:
```diff
@@ -0,0 +1,251 @@
+import { StatusBar } from 'expo-status-bar';
```
general comments:
 
Microsoft license required at the beginning of each file. (can see other files in the repo as an example.)

also maybe the function name itself is sometimes self-explanatory, still can benefit from adding some comments especially for some of the main source files. briefly explain what each function does.

VictorIyke replied:
Should the Microsoft license be at the beginning of every file?

YUNQIUGUO commented on js/app-super-resolution/README.md:
```diff
@@ -0,0 +1,59 @@
+0. Prerequisites
+    1. Install Node.js
+    2. Install Java and OpenJDK
+    3. Install expo
```
node.js/java, etc. maybe common for people,

but for expo fairly new concept, can add link to the name, for people to conveniently click and check. also doesn't hurt to provide links for openjdk, etc. ( for example, is there a specific version required to install?)

VictorIyke replied:
sounds good, adding that

YUNQIUGUO commented on js/app-super-resolution/screens/mobile.tsx:
```diff
@@ -0,0 +1,251 @@
+
+  async function loadModel() {
+    try {
+      const assets = await Asset.loadAsync(require('../assets/super_resnet12.ort'));
```
`../assets/super_resnet12.ort` 

How is the model introduced and used in the example? is it loaded from the assets folder?  (um didn't see it in the folder.)

also in tutorial/readme.md file, would be good to mention models used in this example, limitations about it,  other models can try in the future, etc.

VictorIyke replied:
I just noticed that the model does not get pushed into the repo. So, I added a link to the model in the readme.md file

YUNQIUGUO commented on js/app-super-resolution/screens/mobile.tsx:
```diff
@@ -0,0 +1,251 @@
+  );
+};
+
+
```
nit: remove extra space/lines here

VictorIyke replied:
sure 👍

fs-eire commented:
I have an important question: is this expo project using bare workflow or managed workflow? In other words, after you use the expo-cli to generate the `android` folder, did you do any changes to it?

If no, then you are using managed workflow. The whole `android` folder should not be included in the code base because they are generated.

VictorIyke commented:
@fs-eire 
To answer your question, I made a slight change to a file in the `android` folder and also added two files there. Although I have now removed the android folder from the code, I also explained, in the instructions.md file, how to use the expo-cli to generate the `android` folder and make the small changes.

fs-eire commented on js/app-super-resolution/instructions.md:
```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+
+// Licensed under the MIT license.
```
the copyright banner should only be applied to source code files

VictorIyke replied:
👍 thanks

fs-eire commented on js/app-super-resolution/instructions.md:
```diff
@@ -0,0 +1,107 @@
+
+   2. Install Java (OpenJDK).
+
+      - Download and install [OpenJDK Version 11](https://adoptopenjdk.net/). Extract into the `program files` folder in your local disk.
```
extracting to any place should work (as long as the next step of setting JAVA_HOME is doing correctly).

VictorIyke replied:
Oh, I did not know that. Thanks 👍

fs-eire commented on js/app-super-resolution/instructions.md:
```diff
@@ -0,0 +1,103 @@
+## Steps of building project from scratch
```
The instruction seems only work for Windows system. Is it expected?

VictorIyke replied:
I believe it should only work for Windows. I am not really familiar with MacOS or Linux systems

natke commented:
Superceded by #182
